### PR TITLE
Reorganize VersionManifest

### PR DESCRIPTION
**Pull request by martinemde**
I organized it a bit better and extracted a class for dealing with the ShasumFormat.

### CODE REVIEW

File: app/models/version_manifest.rb [hunk 1]
```diff
@@ -6,38 +6,56 @@ class VersionManifest
   delegate :gem, :fs, to: :contents
 
   def initialize(gem:, number:, platform: nil)
-    @version = platform.present? ? [number, platform].join("-") : number.to_s
-    raise ArgumentError, "version number-platform must be a valid version name" unless @version.match?(Rubygem::NAME_PATTERN)
+    platform = nil if platform == "ruby"
+    @version = platform.present? ? [number, platform.presence].join("-") : number.to_s
```
Comment (segiddins) [inline app/models/version_manifest.rb:10]:
```suggestion
    @version = [number, platform.presence].compact.join("-")
```

File: lib/shasum_format.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#
+module ShasumFormat
+  class ParseError < RuntimeError; end
+
+  # Splits a shasum file format into a Hash of path => checksum
+  #
+  # @param [String] body
+  # @return [Hash] path => checksum
+  def self.parse(body)
+    body.to_s.chomp.split("\n").to_h do |line|
```
Comment (segiddins) [inline lib/shasum_format.rb:40]: `body.to_s.chomp.each_line(chomp: true).to_h`

File: lib/shasum_format.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      line.split("  ").reverse
```
Comment (segiddins) [inline lib/shasum_format.rb:41]: `.split("  ", 2)`

**Review by segiddins — APPROVED** [review]

File: app/models/version_manifest.rb (hunk 1 above)
Comment (martinemde) [reply app/models/version_manifest.rb:10]: oh, of course.

**Pull request merged**